modules/sources/contracts/fetch-erc4626-token-data.ts

```diff
@@ -0,0 +1,110 @@
+import { ViemClient } from '../viem-client';
+import MinimalErc4626Abi from './abis/MinimalERC4626';
+import { fetchErc20Headers } from '.';
+
+export async function fetchErc4626AndUnderlyingTokenData(
+    tokens: { address: string; decimals: number; name: string; symbol: string }[],
+    viemClient: ViemClient,
+): Promise<
+    {
+        address: string;
+        decimals: number;
+        name: string;
+        symbol: string;
+        isErc4626: boolean;
+        underlyingAddress?: string;
+    }[]
+> {
+    const tokenData: {
+        [id: string]: {
+            address: string;
+            decimals: number;
+            name: string;
+            symbol: string;
+            isErc4626: boolean;
+            underlyingAddress?: string;
+        };
+    } = {};
+
+    for (const token of tokens) {
+        let response;
+        try {
+            response = await viemClient.multicall({
+                contracts: [
+                    {
+                        address: token.address as `0x${string}`,
+                        abi: MinimalErc4626Abi,
+                        functionName: 'asset',
+                    },
```

Comment:
it's just the first call is used, are the others to get the rate?

Reply:
the others are just to determine that it is an ERC4626 token

Reply:
ah, it's checking if the whole interface is implemented? try, catch, continue - i see